app/src/main/java/com/cdio/solitaire/imageanalysis/SolitaireAnalysisModel.java

```diff
@@ -0,0 +1,223 @@
+package com.cdio.solitaire.imageanalysis;
+
+import static org.opencv.imgproc.Imgproc.ADAPTIVE_THRESH_MEAN_C;
+import static org.opencv.imgproc.Imgproc.THRESH_BINARY;
+import static java.lang.Double.max;
+import static java.lang.Double.min;
+
+import android.graphics.Bitmap;
+
+import org.opencv.android.Utils;
+import org.opencv.core.Core;
+import org.opencv.core.CvType;
+import org.opencv.core.Mat;
+import org.opencv.core.MatOfPoint;
+import org.opencv.core.MatOfPoint2f;
+import org.opencv.core.Point;
+import org.opencv.core.Rect;
+import org.opencv.core.RotatedRect;
+import org.opencv.core.Size;
+import org.opencv.imgproc.Imgproc;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class SolitaireAnalysisModel {
+
+    // Load in OpenCV library
+    static {
+        System.loadLibrary("opencv_java4");
+    }
+
+    /** Method for extracting suit and rank icon from every card in a Solitaire game image as Bitmaps */
+    public Bitmap[] extractSolitaire(Mat src) {
+        // Convert to BGR colors
+        Imgproc.cvtColor(src, src, Imgproc.COLOR_BGRA2BGR);
+
+        // Extract all 8 cards of the Solitaire game image as and array of ContentNodes
+        ContentNode[] game = extractCards(src, 8);
+
+        // If Solitaire game image is valid
+        if (game != null) {
+            // Solitaire cards are sorted, first by position along the y-axes (talon), and then by position along the x-axes (Columns 1 to 7)
+            Arrays.sort(game, (n1, n2) -> (int) (n1.position.y - n2.position.y));
+            ContentNode talon = game[0];
+            ContentNode[] columns = Arrays.copyOfRange(game,1,8);
+            Arrays.sort(columns, (n1, n2) -> (int) (n1.position.x - n2.position.x));
+
+            // Convert to array of BitMaps and release the Mat objects still in memory
+            Bitmap[] bitmapArr = new Bitmap[8];
+            for (int i = 0; i < 8; i++) {
+                Bitmap bitmap = Bitmap.createBitmap(40, 100, Bitmap.Config.ARGB_8888);
+                if (i == 0) {
+                    Utils.matToBitmap(talon.content, bitmap);
+                    bitmapArr[i] = bitmap;
+                    talon.content.release();
+                } else {
+                    Utils.matToBitmap(columns[i-1].content, bitmap);
+                    bitmapArr[i] = bitmap;
+                    columns[i-1].content.release();
+                }
+            }
+            src.release();
+            return bitmapArr;
+        } else {
+            src.release();
+            System.out.println("No suitable game was found!");
```

Comment:
Maybe add a Log.e or log.w instead or as well.

Reply:
Now corrected to Log.e